outputs.tf

```diff
@@ -50,15 +63,15 @@ output "additional_eni_ids" {
 
 output "ebs_ids" {
   description = "ID of EBSs"
-  value       = "${aws_ebs_volume.default.*.id}"
+  value       = "${join(",", aws_ebs_volume.default.*.id)}"
 }
 
 output "primary_network_interface_id" {
   description = "ID of the instance's primary network interface"
-  value       = "${aws_instance.default.primary_network_interface_id}"
+  value       = "${join(",", aws_instance.default.*.primary_network_interface_id)}"
 }
 
 output "network_interface_id" {
   description = "ID of the network interface that was created with the instance"
-  value       = "${aws_instance.default.network_interface_id}"
+  value       = "${join(",", aws_instance.default.*.network_interface_id)}"
```

Comment:
we should probably output a list here without joining into a string

Reply:
@aknysh as in, `"${split(join(",", aws_instance.default.*.network_interface_id))}"` to get a list, but salso get around the empty array error?

Reply:
Yea, I'm torn here what we should do. Seems like we should be returning lists now, rather than strings albeit that will break backward compatibility.

Reply:
yea, I understand why `,` was used and not `""`.

My question was about two things:
1. We need to output lists because concatenated strings are difficult to use in other modules (we'd have to split them all the time if we wanted to provide a list to other modules)
2. In any case (returning lists or concatenated strings) will break compatibility

Maybe we could create another module, e.g. `terraform-aws-ec2-multi-instance` or `terraform-aws-ec2-instances`, and put all the new changes in there?

`terraform-aws-ec2-instances` would be used as `StaticSet` in k8s and mostly to deploy databases on many instances.

`terraform-aws-ec2-instance` is like `Deployment` and could be used with `ASG` to autoscale.

Also, this PR is too complex now and difficult to follow. Suggest creating new PRs with smaller changes that easier to test (if we want to use this module to apply all the changes).

@osterman  @Jamie-BitFlight  what do you think?